File: python/python-kata/kyu-5/moving_zeros_to_the_end.py
""" Write an algorithm that takes an array and moves all of the zeros to the end, preserving the order of the other elements.

move_zeros([1, 0, 1, 2, 0, 1, 3]) # returns [1, 1, 2, 1, 3, 0, 0] """

def move_zeros(lst):
    non_zeros = []
    zeros = []
    for val in lst:
        if val == 0:
           zeros.append(val)
        else:
            non_zeros.append(val)
    return non_zeros + zeros


test_array = [9, 0, 0, 9, 1, 2, 0, 1, 0, 1, 0, 3, 0, 1, 9, 0, 0, 0, 0, 9]

result = move_zeros(test_array)
print('result: ', result)

File: javascript/js-kata/kyu-5/moving_zeros_to_the_end.js
'use strict';

/* 
Write an algorithm that takes an array and moves all of the zeros to the end, preserving the order of the other elements.

moveZeros([false,1,0,1,2,0,1,3,"a"]) // returns[false,1,1,2,1,3,"a",0,0] 
*/

function moveZeros(arr) {
  const nonZerosArray = [];
  const zerosArray = [];
  
  arr.forEach(ele => {
    if (ele === 0) {
      zerosArray.push(ele);
    } else {
      nonZerosArray.push(ele);
    }
  })

  return [...nonZerosArray, ...zerosArray];
}

const result = moveZeros([false,1,0,1,2,0,1,3,"a"]);
console.log(result);
